django/contrib/postgres/forms/ranges.py

```diff
@@ -1,13 +1,17 @@
 import warnings
 
-from psycopg2.extras import DateRange, DateTimeTZRange, NumericRange
-
 from django import forms
 from django.core import exceptions
 from django.forms.widgets import MultiWidget
 from django.utils.deprecation import RemovedInDjango31Warning
 from django.utils.translation import gettext_lazy as _
 
+try:
+    from psycopg2.extras import DateRange, DateTimeTZRange, NumericRange
+except ImportError:
```

Comment:
I think it's in the "unrelated change" commit. I am a bit in the deep why the tests fail on unrelated database backends.

Reply:
I believe that is because the postgres-specific test modules are still imported, but the tests are skipped. As a result, anything that imports from `django.contrib.postgres` needs to be wrapped to catch `ImportError`, e.g. [this block](https://github.com/django/django/blob/e192223ed996ed30fe83787efdfa7f2be6b1a2ee/tests/postgres_tests/test_apps.py#L7-L15).

Reply:
I tried to do that in `test_constraints.py`, but it doesn't really help.